sbin/prepareWorkspace.sh

```diff
@@ -652,15 +652,31 @@ downloadDevkit() {
   fi
 }
 
-# Download all of the dependencies for OpenJDK (Alsa, FreeType etc.)
+downloadBootJdkIfNeeded () {
+  if [[ "${BUILD_CONFIG[JDK_BOOT_DIR]}" == "download" ]]; then
+    local futureBootDir="${BUILD_CONFIG[WORKSPACE_DIR]}/downloaded-boot-jdk-${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]}"
+    if  [ -e "$futureBootDir" ] ; then
+      echo "Reusing $futureBootDir"
+    else
+      source "$SCRIPT_DIR/common/downloaders.sh"
+      echo "Downloading to $futureBootDir"
+      downloadBootJDK "$(uname -m)" "${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]}" "${futureBootDir}"
+    fi
+    BUILD_CONFIG[JDK_BOOT_DIR]="${futureBootDir}"
+  fi
+}
+
+# Download all of the dependencies for OpenJDK (Alsa, FreeType, boot-jdk etc.)
 downloadingRequiredDependencies() {
   if [[ "${BUILD_CONFIG[CLEAN_LIBS]}" == "true" ]]; then
     rm -rf "${BUILD_CONFIG[WORKSPACE_DIR]}/libs/freetype" || true
-
     rm -rf "${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[WORKING_DIR]}/installedalsa" || true
     rm -rf "${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[WORKING_DIR]}/installedfreetype" || true
+    rm -rf "${BUILD_CONFIG[WORKSPACE_DIR]}/downloaded-boot-jdk-${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]}" || true
```

Comment:
I'm inclined to suggest leaving this in place as a cache for anyone using our scripts since the JDK is quite a big download, but I'm also ok with this.

I was going to comment on the fact that `|| true` is unnecessary with `rm -f` but sine that's also in the previous three lines I'll ignore it ;-)

Reply:
I agree, and I'm keeping it.  It will be removed only if ` if [[ "${BUILD_CONFIG[CLEAN_LIBS]}" == "true" ]]; then`  
oook? Yah, I was wondering  on the double passig rf || true, but left in the style!